featurebyte/common/utils.py

```diff
@@ -137,7 +138,32 @@ def dataframe_from_arrow_stream(buffer: Any) -> pd.DataFrame:
     return reader.read_all().to_pandas()
 
 
-def pa_table_to_record_batches(table: pa.Table) -> Any:
+def _update_batches_for_types(
+    batches: List[pa.RecordBatch], col_name_to_db_var_type: dict[str, DBVarType]
+) -> List[pa.RecordBatch]:
+    # Currently, we only need to perform updates if we have an ARRAY type.
+    if DBVarType.ARRAY not in col_name_to_db_var_type.values():
+        return batches
+
+    output_list = []
+    for batch in batches:
+        curr_df = batch.to_pandas()
+        for col_name, db_var_type in col_name_to_db_var_type.items():
+            if DBVarType.ARRAY == db_var_type:
+                # Check if column has elements of string type
+                is_string_series = curr_df[col_name].apply(lambda x: isinstance(x, str))
```

Comment:
This could be a bit slow. Could we use something like
```
is_string_series = (curr_df[col_name] == curr_df[col_name].astype(str)).all()
```

Reply:
Can also consider using [`is_string_dtype`](https://pandas.pydata.org/docs/reference/api/pandas.api.types.is_string_dtype.html) which performs the check in cython and should be more efficient.